src/Microsoft.ML.OnnxTransform/DnnImageFeaturizerTransform.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Core.Data;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.Internal.Utilities;
+using Microsoft.ML.StaticPipe;
+using Microsoft.ML.StaticPipe.Runtime;
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.ML.Transforms
+{
+    // This is a helper class that is required to use the DnnImageFeaturizer estimator.
+    // Note that by default, it is not usable as it does not have any valid methods that return an EstimatorChain that
+    // is used by the DnnImageFeaturizeEstimator.
+    // In order to use this, at least one model project with the corresponding extension methods must by included.
+    // See Microsoft.ML.DNNImageFeaturizer.ResNet18 for an example.
+    public sealed class DnnImageModelSelector
+    {
+    }
+
+    // The Dnn Image Featurizer is just a wrapper around two OnnxTransforms with present pretrained DNN models.
+    // Note that because of this, it only works on Windows machines as that is a constraint of the OnnxTransform.
+    public sealed class DnnImageFeaturizerEstimator : IEstimator<TransformerChain<OnnxTransform>>
```

Comment:
>IEstimator [](start = 54, length = 10)

ITransformer?

Reply:
No? This is an estimator?

---
In reply to: [232402105](https://github.com/dotnet/machinelearning/pull/1447#discussion_r232402105) [](ancestors = 232402105)